Issues compiling NPS on Ubuntu 22.04

I'm testing on Ubuntu 22.04 the changes made by @Fabien-B in #2922 and I experienced a problem. Since I believe it is not directly related to the changes he has made in this PR, I decided to open this issue.

When I try to compile any airframe with NPS as target, the compiler drops some multiple-definition errors like this:

/usr/bin/ld: 
/home/ubuntu-22/Workspace/paparazzi/var/aircrafts/MYAC/nps/nps/nps_ivy.o:
/home/ubuntu-22/Workspace/paparazzi/sw/simulator/nps/nps_main.h:28: 
multiple definition of `fdm_mutex'; 
/home/ubuntu-22/Workspace/paparazzi/var/aircrafts/MYAC/nps/nps/nps_flightgear.o:
/home/ubuntu-22/Workspace/paparazzi/sw/simulator/nps/nps_main.h:28: 
first defined here

Could It be related to a compiler flag?

--- a/sw/simulator/nps/nps_ivy.c
+++ b/sw/simulator/nps/nps_ivy.c
@@ -26,6 +26,7 @@
 #include "modules/gps/gps.h"
 #endif
 
+bool nps_ivy_send_world_env = false;
 pthread_t th_ivy_main; // runs main Ivy loop
 static MsgRcvPtr ivyPtr = NULL;
 static int seq = 1;

--- a/sw/simulator/nps/nps_ivy.h
+++ b/sw/simulator/nps/nps_ivy.h
@@ -4,7 +4,7 @@
 #include "nps_fdm.h"
 #include "nps_sensors.h"
 
-bool nps_ivy_send_world_env;
+extern bool nps_ivy_send_world_env;
 
 extern void nps_ivy_init(char *ivy_bus);
 extern void nps_ivy_display(struct NpsFdm* fdm_ivy, struct NpsSensors* sensors_ivy);

--- a/sw/simulator/nps/nps_main.h
+++ b/sw/simulator/nps/nps_main.h
@@ -21,30 +21,30 @@ void clock_get_current_time(struct timespec *ts);
 #define DISPLAY_DT (1./30.)
 #define HOST_TIMEOUT_MS 40
 
-pthread_t th_flight_gear; // sends/receives flight gear packets
-pthread_t th_display_ivy; // sends Ivy messages
-pthread_t th_main_loop; // handles simulation
+extern pthread_t th_flight_gear; // sends/receives flight gear packets
+extern pthread_t th_display_ivy; // sends Ivy messages
+extern pthread_t th_main_loop; // handles simulation
 
-pthread_mutex_t fdm_mutex; // mutex for fdm data
+extern pthread_mutex_t fdm_mutex; // mutex for fdm data
 
-int pauseSignal; // for catching SIGTSTP
+extern int pauseSignal; // for catching SIGTSTP
 
-bool nps_main_parse_options(int argc, char **argv);
+extern bool nps_main_parse_options(int argc, char **argv);
 
-int nps_main_init(int argc, char **argv);
-void nps_radio_and_autopilot_init(void);
-void nps_main_run_sim_step(void);
-void nps_set_time_factor(float time_factor);
+extern int nps_main_init(int argc, char **argv);
+extern void nps_radio_and_autopilot_init(void);
+extern void nps_main_run_sim_step(void);
+extern void nps_set_time_factor(float time_factor);
 
-void* nps_main_loop(void* data __attribute__((unused)));
-void* nps_flight_gear_loop(void* data __attribute__((unused)));
-void* nps_main_display(void* data __attribute__((unused)));
+extern void* nps_main_loop(void* data __attribute__((unused)));
+extern void* nps_flight_gear_loop(void* data __attribute__((unused)));
+extern void* nps_main_display(void* data __attribute__((unused)));
 
-void tstp_hdl(int n __attribute__((unused)));
-void cont_hdl(int n __attribute__((unused)));
+extern void tstp_hdl(int n __attribute__((unused)));
+extern void cont_hdl(int n __attribute__((unused)));
 
-double time_to_double(struct timeval *t);
-double ntime_to_double(struct timespec *t);
+extern double time_to_double(struct timeval *t);
+extern double ntime_to_double(struct timespec *t);
 
 void nps_update_launch_from_dl(uint8_t value);
 
@@ -67,6 +67,6 @@ struct NpsMain {
   bool nodisplay;
 };
 
-struct NpsMain nps_main;
+extern struct NpsMain nps_main;
 
 #endif /* NPS_MAIN_H */

--- a/sw/simulator/nps/nps_main_common.c
+++ b/sw/simulator/nps/nps_main_common.c
@@ -30,6 +30,13 @@
 
 #include "nps_ivy.h"
 
+pthread_t th_flight_gear;
+pthread_t th_display_ivy;
+pthread_t th_main_loop;
+pthread_mutex_t fdm_mutex;
+int pauseSignal;
+struct NpsMain nps_main;
+
 #ifdef __MACH__
 pthread_mutex_t clock_mutex; // mutex for clock
 void clock_get_current_time(struct timespec *ts)

